### ISSUE TITLE
Could not parse @Component() metadata error on templates with `\`

### ISSUE THREAD

**Issue by laurentgoudet** [angular-eslint/angular-eslint#207]
The template parser currently fails on templates with `\` in them with, e.g. `<span>\${{ category.cost }} USD</span>`.

Stack trace:

```
Error: Debug Failure. False expression: position cannot precede the beginning of the file
    at computeLineOfPosition (/foo/node_modules/typescript/lib/typescript.js:8549:22)
    at computeLineAndCharacterOfPosition (/foo/node_modules/typescript/lib/typescript.js:8527:26)
    at Object.getLineAndCharacterOfPosition (/foo/node_modules/typescript/lib/typescript.js:8568:16)
    at SourceFileObject.getLineAndCharacterOfPosition (/foo/node_modules/typescript/lib/typescript.js:138709:23)
    at preprocess (/foo/node_modules/@angular-eslint/eslint-plugin-template/dist/index.js:1:1492)
    at Linter._verifyWithProcessor (/foo/node_modules/eslint/lib/linter/linter.js:1292:30)
    at Linter._verifyWithConfigArray (/foo/node_modules/eslint/lib/linter/linter.js:1264:25)
    at Linter.verify (/foo/node_modules/eslint/lib/linter/linter.js:1226:25)
    at Linter.verifyAndFix (/foo/node_modules/eslint/lib/linter/linter.js:1416:29)
    at verifyText (/foo/node_modules/eslint/lib/cli-engine/cli-engine.js:224:48)
preprocess: ERROR could not parse @Component() metadata /<file>
```

Error message is the same https://github.com/angular-eslint/angular-eslint/issues/185, but root cause seems different.

**Comment by laurentgoudet** [comment]
So I've added a failing test & some debug logs in https://github.com/laurentgoudet/angular-eslint/commit/3c32b7bca00446c932ecd10652d96679cf18a765, but I'm not sure how to fix it.

Basically the `template` initializer node `text` value feels broken, as the escaped characters have been removed (e.g. `'It's free'` below is invalid), but at the same time this is what `preprocessComponentFile()` (wrongly?) returns.

```
      text: '\n        <input [value]="${{this.aCost}} USD"></input>\n' +
        `        <input [value]="'It's free'"></input>\n      `,
      rawText: '\n        <input [value]="\\${{this.aCost}} USD"></input>\n' +
        `        <input [value]="'It\\'s free'"></input>\n      `,
```

Any thoughts on how this could be addressed? For now I've worked around it using `<ng-container>` wrappers but escaping characters are valid in Angular templates so ideally the template parser should support them.

Additional logs:

```
$ npx jest -c packages/eslint-plugin-template/jest.config.js packages/eslint-plugin-template/tests/processors.test.ts
...
console.log
    TokenObject {
      pos: 139,
      end: 251,
      flags: 0,
      modifierFlagsCache: 0,
      transformFlags: 256,
      parent: NodeObject {
        pos: 115,
        end: 251,
        flags: 0,
        modifierFlagsCache: 0,
        transformFlags: 256,
        parent: NodeObject {
          pos: 78,
          end: 316,
          flags: 0,
          modifierFlagsCache: 0,
          transformFlags: 256,
          parent: [NodeObject],
          kind: 197,
          properties: [Array],
          multiLine: true
        },
        kind: 285,
        decorators: undefined,
        modifiers: undefined,
        symbol: undefined,
        localSymbol: undefined,
        locals: undefined,
        nextContainer: undefined,
        name: IdentifierObject {
          pos: 115,
          end: 138,
          flags: 0,
          modifierFlagsCache: 0,
          transformFlags: 0,
          parent: [Circular],
          kind: 78,
          originalKeywordKind: undefined,
          escapedText: 'template'
        },
        initializer: [Circular],
        questionToken: undefined,
        exclamationToken: undefined
      },
      kind: 14,
      text: '\n        <input [value]="${{this.aCost}} USD"></input>\n' +
        `        <input [value]="'It's free'"></input>\n      `,
      rawText: '\n        <input [value]="\\${{this.aCost}} USD"></input>\n' +
        `        <input [value]="'It\\'s free'"></input>\n      `,
      templateFlags: 0
    }

      at Object.preprocessComponentFile [as preprocess] (src/processors.ts:147:15)

  console.log

                import {Component} from '@angular/core';

                @Component({
                  selector: 'app-root',
                  template: `
            <input [value]="\${{this.aCost}} USD"></input>
            <input [value]="'It\'s free'"></input>
          `,
                  styleUrls: ['./app.component.scss']
                })
                export class AppComponent {
                  public aCost = 20;
                }

      at Object.preprocessComponentFile [as preprocess] (src/processors.ts:151:15)

  console.log

            <input [value]="${{this.aCost}} USD"></input>
            <input [value]="'It's free'"></input>

      at Object.preprocessComponentFile [as preprocess] (src/processors.ts:152:15)

  console.log
    [ -1, 106 ]

      at Object.preprocessComponentFile [as preprocess] (src/processors.ts:153:15)

  console.log
    Error: Debug Failure. False expression: position cannot precede the beginning of the file
        at computeLineOfPosition (/foo/angular-eslint/node_modules/typescript/lib/typescript.js:8934:22)
        at computeLineAndCharacterOfPosition (/foo/angular-eslint/node_modules/typescript/lib/typescript.js:8912:26)
        at Object.getLineAndCharacterOfPosition (/foo/angular-eslint/node_modules/typescript/lib/typescript.js:8953:16)
        at SourceFileObject.Object.<anonymous>.SourceFileObject.getLineAndCharacterOfPosition (/foo/angular-eslint/node_modules/typescript/lib/typescript.js:143309:23)
        at Object.preprocessComponentFile [as preprocess] (/foo/angular-eslint/packages/eslint-plugin-template/src/processors.ts:160:29)
        at Object.<anonymous> (/foo/angular-eslint/packages/eslint-plugin-template/tests/processors.test.ts:331:47)
        at Object.asyncJestTest (/foo/angular-eslint/node_modules/jest-jasmine2/build/jasmineAsyncInstall.js:106:37)
        at /foo/node_modules/jest-jasmine2/build/queueRunner.js:45:12
        at new Promise (<anonymous>)
        at mapper (/foo/angular-eslint/node_modules/jest-jasmine2/build/queueRunner.js:28:19)
```

**Comment by marcinowski** [comment]
Hi, I stumbled upon this issue too in https://github.com/eslint/eslint/issues/14066. Adding to the @laurentgoudet 's contest my usecase was having a conditional in the component's template
```
@Component({
  selector: 'app-component',
  template: ``
	<component
		[prop]="true ? 'test for \\'@\\' escaping' : 'false'"
	></component>
`)};
```
where the escape characters are needed because of using all the available quoting. Doing this results in the same error as described above.

**Comment by JamesHenry** [comment]
I'm sorry this took such a long time to put right folks, but I believe I finally figured out a solution for this.

It should be available in the latest canary release once master builds and released in a specific version on the latest tag soon

**Comment by JamesHenry** [comment]
It should be available now in `12.5.0`
